File: Payhub - SDK/ruby_SDK/PayHubSDK/lib/PayHubSDK/com/payhub/ws/api/abstract_info.rb
class AbstractInfo
  include JsonSerializer
  ATTRS=[:version,:createdAt,:lastModified,:createdBy,:lastModifiedBy,:metaData]
  attr_accessor :url,:transactionManager,:errors,*ATTRS

  def getDataByID(id)
    url = @transactionManager.url+getUrlForTransactionType(@transactionType)+id.to_s
    request = @transactionManager.setHeadersGet(url, @transactionManager.token)
    json=@transactionManager.doGet(request)
    return nil if json==nil or json==""
    convertData(json_encode(json))
    convertAbstractData(json_encode(json))
  end

  def getDataByTransaction(type,transactionId)
    url=nil
    if TransactionType::CardData==@transactionType
    url = @transactionManager.url+getUrlForTransactionType(type)+transactionId.to_s+"/card_data"
    else
        url = @transactionManager.url+getUrlForTransactionType(type)+transactionId.to_s+"/"+getUrlForTransactionType(@transactionType)
    end
    #request = @transactionManager.setHeadersGet(url, @transactionManager.token)
    json=@transactionManager.doGet(url, @transactionManager.token)
    return nil if json==nil or json==""
    result=JSON.parse(json)
    errors ||= Array.new
    if result.include?('errors')
      result['errors'].each do |error|
        response_tmp = Errors.from_json(JSON.generate(error))
        errors.push(response_tmp)
      end
      @errors=errors
      return @errors
    end
    convertData(json)
    convertAbstractData(json)
    end
  def convertAbstractData(json)
    obj = JSON.parse(json)
    if (obj.key?("version"))
      @version=obj['version']
    end
    if (obj.key?("createdAt"))
      @version=obj['createdAt']
    end
    if (obj.key?("lastModified"))
      @version=obj['lastModified']
    end
    if (obj.key?("createdBy"))
      @version=obj['createdBy']
    end
    if (obj.key?("lastModifiedBy"))
      @version=obj['lastModifiedBy']
    end
    if (obj.key?("metaData"))
      @version=obj['metaData']
    end
  end

end

File: Payhub - SDK/php_SDK/com/payhub/ws/api/AbstractInfo.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 09:26
 */
abstract class AbstractInfo
{
    protected $version;
    protected $createdAt;
    protected $lastModified;
    protected $createdBy;
    protected $lastModifiedBy;
    protected $metaData;
    protected $transactionManager;
    protected $transactionTsype;

    /**
     * AbstractInfo constructor.
     * @param $transactionManager
     */
    public function __construct($transactionManager)
    {
        if(!is_null($transactionManager)) {
            $this->transactionManager = $transactionManager;
        }
        $this->transactionType=TransactionType::Merchant;
    }
    public function getDataByID($id){
        $url = $this->transactionManager->getUrl().$this->getUrlForTransactionType($this.transactionType).$id;
        $request = $this->transactionManager->setHeadersGet($url, $this->transactionManager->getToken());
        $json=$this->transactionManager->doGet($request);
        $this->convertData(json_encode($json));
        $this->convertAbstractData(json_encode($json));
    }
    public function getDataByTransaction($type,$transactionId){
        $url=null;
		if(TransactionType::CardData==$this->transactionType){
            $url = $this->transactionManager->getUrl().$this->getUrlForTransactionType($type).$transactionId."/card_data";
        }else{
            $url = $this->transactionManager->getUrl().$this->getUrlForTransactionType($type).$transactionId."/".$this->getUrlForTransactionType($this->transactionType);
        }
		$request = $this->transactionManager->setHeadersGet($url, $this->transactionManager->getToken());
        $json=$this->transactionManager->doGet($request);
        $this->convertData(json_encode($json));
        $this->convertAbstractData(json_encode($json));
	}
	public abstract function convertData($json);

	public  function convertAbstractData($json){
        $obj = json_decode($json);
        if(array_key_exists('version',$obj))
            $this->version = $obj->{'version'};
        if(array_key_exists('createdAt',$obj))
    	    $this->createdAt=$obj->{'createdAt'};
        if(array_key_exists('lastModified',$obj))
            $this->lastModified=$obj->{'lastModified'};
        if(array_key_exists('createdBy',$obj))
            $this->createdBy=$obj->{'createdBy'};
        if(array_key_exists('lastModifiedBy',$obj))
            $this->lastModifiedBy=$obj->{'lastModifiedBy'};
        if(array_key_exists('metaData',$obj))
            $this->metaData=$obj->{'metaData'};
	}
	public abstract function getUrlForTransactionType($type);
}